File: components/core/src/test/resources/cucumber/runtime/rhino/stepdefs.js
World(function() {

});

Given(/^I have (\d+) cukes in my belly$/, function(n) {
    this.cukes = n;
});

Then(/^there are (\d+) cukes in my belly$/, function(n) {
    if (this.cukes != n) {
        throw "Expected " + n + ", but got " + this.cukes;
    }
});

File: components/clojure/src/test/resources/cucumber/runtime/clojure/stepdefs.clj
(use 'cucumber.runtime.clojure.belly)

(Given #"^I have (\d+) cukes in my belly$"
  #(eat (Float. %1)))

(When #"^there are (\d+) cukes in my belly$"
  (fn [expected]
    (assert (= (last-meal) (Float. expected)))))